challenges/08-coding-interview-prep/rosetta-code.json

```diff
@@ -12434,6 +12434,1022 @@
+      ],
+      "tests": [
+        {
+          "text": "'<code>pascalMat</code> should be a function.'",
+          "testString":
+            "assert(typeof pascalMat=='function','<code>pascalMat</code> should be a function.');"
+        },
+        {
+          "text":
+            "'<code>pascalMat(\"lower\",3)</code> should return a array.'",
+          "testString":
+            "assert(Array.isArray(pascalMat(\"lower\",3)),'<code>pascalMat(\"lower\",3)</code> should return a array.');"
+        },
+        {
+          "text":
+            "'<code>pascalMat(\"lower\",3)</code> should return <code>'+JSON.stringify(results[0])+'</code>.'",
+          "testString":
+            "assert.deepEqual(pascalMat(\"lower\",3),results[0],'<code>pascalMat(\"lower\",3)</code> should return <code>'+JSON.stringify(results[0])+'</code>.');"
```

Comment:
@scissorsneedfoodtoo The single quotes are because of the above lines. Many of the challenges' test cases are stored in arrays. To display them, `JSON.stringify(...)` have been used. The result needs to be concatenated. As you can see in the above example, If we remove the single quotes, the expression's quotes becomes unbalanced as the concatenation uses single quotes. We can either replace the `JSON.stringify(...)` by including the raw arrays directly into the strings or we can implement a mechanism that evaluates the strings as JS expressions first so that `JSON.stringify(...)` is evaluated and then the resultant string is displayed in th UI.

Reply:
Ah, I see. Thanks for the thorough explanation! This method works totally fine for the console window / testString, but not for the test suite window for some reason. If you take a look at the left side of the screen you'll see what I mean:

![image](https://user-images.githubusercontent.com/2051070/44456060-9866fb80-a63a-11e8-9f69-2c817e3a5111.png)

Do you know of a way we can get those strings to render properly? Been playing with it a bit on my end and wasn't able to get it working.

Reply:
Yeah, I saw that earlier. I will check how the text and testString are rendered. Do you know of the location of the code for this?

Reply:
I was just looking more into this. [Here's](https://github.com/freeCodeCamp/learn/blob/59a4bb913c64e27839814f4f84b932d3031f09ae/src/templates/Challenges/components/Test-Suite.js) the file that handles rendering the test texts to the page.

Reply:
Thanks!! I will look into it. If I find a solution, I will make a separate pull request.

Reply:
Okay! Sounds great. I'll keep messing with it, too. Worst case scenario we can go back and just edit the texts to show the arrays.